controllers/users.js

```diff
@@ -0,0 +1,97 @@
+const User = require( '../models/schemas/user' );
+const config = require( '../models/config' );
+
+exports.getUsers = ( req, res, next ) => {
+	User.find( {}, ( err, users ) => {
+		if ( err ) return next( err );
+		return res.json( users );
+	});
+}
+
+exports.createUser = ( req, res, next ) => {
+	if ( typeof req.body.email !== 'string' )
+		return res.status( 400 ).send( 'No email');
+	if ( typeof req.body.password !== 'string' )
+		return res.status( 400 ).send( 'No password' );
+	if ( typeof req.body.username !== 'string' )
+		return res.status( 400 ).send( 'No username' );
+
+	let userData = {};
+
+	if ( req.body.firstName && typeof req.body.firstName === 'string' )
```

Comment:
This is too much redundant code, create an array of required parameters and loop over them using the same code. ie:

```
let requiredParams = [ 'firstName', 'lastName' , ... ],
error = false;

requiredParams.forEach( ( param ) => {
  if ( error ) return;
  // TODO: add proper form validation somehow
  if ( typeof req.body.username !== 'string' || req.body.username )
    error = 'Missing required' + param;
} );

if ( error ) return res.status( 400 ).send( error );
```

Reply:
done